src/resources/views/crud/inc/form_fields_script.blade.php

```diff
@@ -0,0 +1,162 @@
+<script>
+    /**
+     * A front-end representation of a Backpack field, with its main components.
+     *
+     * Makes it dead-simple for the developer to perform the most common
+     * javascript manipulations, and makes it easy to do custom stuff
+     * too, by exposing the main components (name, wrapper, input).
+     */
+    class CrudField {
+        constructor(name) {
+            this.name = name;
+            this.wrapper = $(`[bp-field-name*="${name}"][bp-field-wrapper]`).first();
+
+            if (this.wrapper.length === 0) {
+                console.error(`CrudField error! Could not select WRAPPER for "${this.name}"`);
+            }
+
+            this.type = this.wrapper.attr('bp-field-type');
+            this.$input = this.mainInput;
+            this.input = this.$input[0];
+
+            // Validate that the field has been found
+            if(!this.input) {
+                console.error(`CrudField error! Could not select INPUT for "${this.name}"`);
+            }
+        }
+
+        get value() {
+            return this.$input.val();
+        }
+
+        get mainInput() {
+            let input;
+
+            // search input in ancestors
+            input = this.wrapper.closest('[bp-field-main-input]');
+
+            // search input in children
+            if (input.length === 0) {
+                input = this.wrapper.find('[bp-field-main-input]');
+            }
+
+            // if no bp-field-main-input has been declared in the field itself, try to find an input with that name inside wraper
+            if (input.length === 0) {
+                input = this.wrapper.find(`input[bp-field-name="${this.name}"], textarea[bp-field-name="${this.name}"], select[bp-field-name="${this.name}"]`).first();
+            }
+
+            // if nothing works, use the first input found in field wrapper.
+            if(input.length === 0) {
+                input = this.wrapper.find('input, textarea, select').first();
+            }
+
+            return input;
+        }
+
+        onChange(closure) {
+            const bindedClosure = closure.bind(this);
+            const fieldChanged = (event, values) => bindedClosure(this, event, values);
+
+            if(this.isSubfield) {
+                window.crud.subfieldsCallbacks = window.crud.subfieldsCallbacks ?? [];
+                window.crud.subfieldsCallbacks[this.subfieldHolder] = window.crud.subfieldsCallbacks[this.subfieldHolder] ?? [];
+                if(!window.crud.subfieldsCallbacks[this.subfieldHolder].some( callbacks => callbacks['fieldName'] === this.name )) {
+                    window.crud.subfieldsCallbacks[this.subfieldHolder].push({fieldName: this.name, closure: closure, field: this});
+                }
+                return this;
+            }
+
+            if(['text', 'textarea'].includes(this.input.getAttribute('type'))) {
+                this.input?.addEventListener('input', fieldChanged, false);
+            }
```

Comment:
Found a bug that might be related to this. In my tests, inside the SIMPLE tab, only the "text" field now triggers the `change()` event after each key press. The rest of the fields, even tough they are technically INPUT or TEXTAREA... do not, they only trigger it on focusout:

https://user-images.githubusercontent.com/1032474/172439286-58fdaf76-80d1-4304-8740-ef6c46d6d45f.mp4

Reply:
@tabacitu can't reproduce. I am on the three main branches with all last commits: https://recordit.co/sYesm7qSnk

Reply:
I will post what I posted on discord.

We need to add `?` into this input, or it will raise an error when the field does not exist.

![image](https://user-images.githubusercontent.com/7188159/172443391-25f4925e-4c50-4bab-b0a4-04af7c8100c8.png)

Sorry I did the sarcastic comment about the conditional if, it migth seem that I was kidding and not talking serious about using `this.input?` ! My bad. 

It was serious! 😈

Reply:
> We need to add ? into this input, or it will raise an error when the field does not exist.

Ok done!

Reply:
About the change not triggered on keyup... I couldn't reproduce now either... 😅 But I have PROOF it was happening. Yeah perhaps some of the merges fixed it. Or I had changes that broke it... 🤷‍♂️ Either way, working now 🎉  Sorry for the false alarm.